src/index.ts

```diff
@@ -1,69 +1,215 @@
 import {
   JupyterFrontEnd,
-  JupyterFrontEndPlugin
+  JupyterFrontEndPlugin,
+  ILayoutRestorer
 } from '@jupyterlab/application';
 import { deployAppIcon } from './icons';
-import { DocumentWidget } from '@jupyterlab/docregistry';
+import { WidgetTracker, IFrame, MainAreaWidget } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
 
+const DEFAULT_BASE_URL = '/services/japps/create-app';
 namespace CommandIDs {
   /**
-   * Opens the URL to deploy the application with pre-populated fields
+   * Opens the URL for deploying an app with pre-populated fields.
    */
   export const deployApp = 'jhub-apps:deploy-app';
 }
 
 interface IDeployAppArgs {
   /**
-   * The origin of the command e.g. main-menu, context-menu, etc.
+   * The origin of the command e.g. main-menu, context-menu, etc.,
+   * which determines the icon and label to display.
    */
   origin?: string;
+  /**
+   * The base URL of the app deployment form.
+   */
+  baseUrl?: string;
+  /**
+   * Additional query parameters to be appended to the baseUrl.
+   */
+  queryParameters?: Record<string, string>;
```

Comment:
We could probably add tighter typing here to guard against future typos, like:

```suggestion
  queryParameters?: {
    headless: string;
    filepath: string
  };
```

Or that could even be an interface to allow for reuse later down here.